src/ctapipe/irf/irfs.py

```diff
@@ -0,0 +1,316 @@
+"""Components to generate IRFs"""
+from abc import abstractmethod
+
+import astropy.units as u
+import numpy as np
+from astropy.io.fits import BinTableHDU
+from astropy.table import QTable
+from pyirf.io import (
+    create_aeff2d_hdu,
+    create_background_2d_hdu,
+    create_energy_dispersion_hdu,
+    create_psf_table_hdu,
+)
+from pyirf.irf import (
+    background_2d,
+    effective_area_per_energy,
+    effective_area_per_energy_and_fov,
+    energy_dispersion,
+    psf_table,
+)
+from pyirf.simulations import SimulatedEventsInfo
+
+from ..core.traits import AstroQuantity, Float, Integer
+from .binning import FoVOffsetBinsBase, RecoEnergyBinsBase, TrueEnergyBinsBase
+
+
+class PsfMakerBase(TrueEnergyBinsBase):
+    """Base class for calculating the point spread function."""
+
+    def __init__(self, parent=None, **kwargs):
+        super().__init__(parent=parent, **kwargs)
+
+    @abstractmethod
+    def make_psf_hdu(self, events: QTable, extname: str = "PSF") -> BinTableHDU:
+        """
+        Calculate the psf and create a fits binary table HDU in GAD format.
+
+        Parameters
+        ----------
+        events: astropy.table.QTable
+            Reconstructed events to be used.
+        extname: str
+            Name for the BinTableHDU.
+
+        Returns
+        -------
+        BinTableHDU
+        """
+
+
+class BackgroundRateMakerBase(RecoEnergyBinsBase):
+    """Base class for calculating the background rate."""
+
+    def __init__(self, parent=None, **kwargs):
+        super().__init__(parent=parent, **kwargs)
+
+    @abstractmethod
+    def make_bkg_hdu(
+        self, events: QTable, obs_time: u.Quantity, extname: str = "BACKGROUND"
+    ) -> BinTableHDU:
+        """
+        Calculate the background rate and create a fits binary table HDU
+        in GAD format.
+
+        Parameters
+        ----------
+        events: astropy.table.QTable
+            Reconstructed events to be used.
+        obs_time: astropy.units.Quantity[time]
+            Observation time. This must match with how the individual event
+            weights are calculated.
+        extname: str
+            Name for the BinTableHDU.
+
+        Returns
+        -------
+        BinTableHDU
+        """
+
+
+class EnergyMigrationMakerBase(TrueEnergyBinsBase):
+    """Base class for calculating the energy migration."""
+
+    energy_migration_min = Float(
+        help="Minimum value of Energy Migration matrix",
+        default_value=0.2,
+    ).tag(config=True)
+
+    energy_migration_max = Float(
+        help="Maximum value of Energy Migration matrix",
+        default_value=5,
+    ).tag(config=True)
+
+    energy_migration_n_bins = Integer(
+        help="Number of bins in log scale for Energy Migration matrix",
+        default_value=30,
+    ).tag(config=True)
+
+    def __init__(self, parent=None, **kwargs):
+        super().__init__(parent=parent, **kwargs)
+        self.migration_bins = np.linspace(
```

Comment:
I'll fix it. And I also think renaming it to energy dispersion would be good to avoid confusion with the naming in pyirf (e.g. the `energy_dispersion_to_migration` function).

Reply:
No, this variable is called migration also in pyirf, e.g. here:
https://pyirf.readthedocs.io/en/latest/api/pyirf.irf.energy_dispersion.html#pyirf.irf.energy_dispersion

(migration bins)

Reply:
Ok, but I meant renaming `EnergyMigrationMaker` into `EnergyDispersionMaker` since it returns the dispersion matrix (true energy vs (true - reco) / true) and not the migration matrix (true energy vs reco energy).

Reply:
Ah, yes! I thought you meant renaming the variable